R/combinevariablesetsasbinary.R

```diff
@@ -151,8 +161,45 @@ questionToBinary <- function(x) {
         binary.version <- AsNumeric(x, binary = TRUE, name = levels(x))
         colnames(binary.version) <- levels(x) 
         binary.version[is.na(x), ] <- NA
+        attr(binary.version, "originalquestiontype") <- "Pick One"
         return(binary.version)
     }
 
     stop("Unsupported data type: ", question.type)
 }
+
+# Function to expand the number of columns in the binary data
+# when there are fewer columns than expected. expected.columns
+# should be a vector of column names. 
+fillInCategoriesWhenNotPresent <- function(binary.data, expected.columns, pick.any.all.missing = TRUE) {
+    
+    current.colnames <- colnames(binary.data)
+
+    if (all(expected.columns %in% colnames(current.colnames)))
+        return(binary.data)
+
+    new.colnames <- expected.columns[! expected.columns %in% current.colnames]
+    new.data <- matrix(FALSE, nrow = nrow(binary.data), ncol = length(new.colnames))
+    colnames(new.data) <- new.colnames
+
+
+    # Missing data rule
+    # For data which was originally mutually-exclusive,
+    # cases are assigned missing values in the new columns
+    # when the case has missing data in the existing columns.
+    # In this case the row will always be entirely missing
+    # or entirely non-missing.
+    # For data which was already binary, new columns should be
+    # entirely missing unless overridden by the argument.
+    n.missing.per.case <- SumEachRow(is.na(binary.data))
+    missing.in.new.data = rep(TRUE, nrow(binary.data))
+    if (attr(binary.data, "originalquestiontype") == "Pick One" | !pick.any.all.missing) {
```

Comment:
Should use `||` instead of `|` although there is no danger of vector inputs (currently)